### DISCUSSION TITLE
ecoMAX360i sensors and heat pump controls

### DISCUSSION THREAD

**Discussion by dschense** [jontofront/ecoNET-300-Home-Assistant-Integration#253] [Ideas]
### The request

These are my sensors I get on econet
![image](https://github.com/user-attachments/assets/1ca1ec4f-606c-4fad-9f7d-6efd5889dcff)

Iam not that good in the API Stuff, but maybe this one helps to get some of these into the project

These 2 are for controlling Temp of Circuit 3 (floor heating) and DHW (hot water)

Circuit 3 Set Temp in Celcius:
```
Edit: /service/newParam
https://www.econet24.com/service/newParam?uid=XYZ&newParamName=361&newParamValue=32&_=UID
http://IP-OF-econet/econet/newParam?newParamName=361&newParamValue=32&_=UID

{
    "paramName": "361",
    "paramValue": 32,
    "result": "OK"
}
```
DHW set Temp in Celcius:
```
Edit: /service/newParam
https://www.econet24.com/service/newParam?uid=XYZ&newParamName=103&newParamValue=45&_=UID
http://IP-OF-econet/econet/newParam?newParamName=103&newParamValue=45&_=UID

{
    "paramName": "103",
    "paramValue": 45,
    "result": "OK"
}
```

Maybe someone can help me to find more to get the other ones included.

**Comment by Rumb0ldt** [comment]
In the older integration for FoxAir (https://github.com/zaapp/foxair-ha/tree/master) there are further controls with their keys: https://github.com/zaapp/foxair-ha/blob/master/custom_components/econet300/number.py.

![grafik](https://github.com/user-attachments/assets/faca2907-fd9c-416a-aa68-e78330daaeb0)

Temperature circuit 1 – NIGHT - {"paramName": "239", "paramValue":18, "result":"OK"}

Temperature circuit 1 – TAG - {"paramName": "238", "paramValue":24, "result":"OK"}

Buffer target temperature - {"paramName": "183", "paramValue":39, "result":"OK"}

Temperature circuit 2 - night - {"paramName": "289", "paramValue":15, "result":"OK"}

Temperature circuit 2 - day - {"paramName": "288", "paramValue":21.5, "result":"OK"}

My aim is to extract also the other attributes and keys from the older code for the FoxAir heatpumps.

**Comment by jontofront** [comment]
Whet model ecomax you have?

**Comment by Rumb0ldt** [comment]
> Whet model ecomax you have?

I am using a ecoMAX360i.

**Comment by jontofront** [comment]
> > Whet model ecomax you have?
> 
> I am using a ecoMAX360i.

We need to find the API endpoint for editable parameters.
For example, I have the following endpoints:

```
http://econet_local_ip/econet/regParams
http://econet_local_ip/econet/regParamsData
http://econet_local_ip/econet/sysParams
http://econet_local_ip/econet/rmCurrentDataParams
http://econet_local_ip/econet/rmCurrentDataParamsEdits
```

The endpoint for editable values appears to be:

`http://econet_local_ip/econet/rmCurrentDataParamsEdits`

Let me know if you need help explaining what each endpoint does or how to use them.

**Comment by Rumb0ldt** [comment]
> ```
> http://econet_local_ip/econet/regParams ✔️ 
> http://econet_local_ip/econet/regParamsData ❌ `{"error":"'CommThread' object has no attribute 'getCurrentParamsValues'"}`
> http://econet_local_ip/econet/sysParams ✔️ 
> http://econet_local_ip/econet/rmCurrentDataParams ❌  `{"error":"'CommThread' object has no attribute 'getRemoteMenuParams'"}`
> http://econet_local_ip/econet/rmCurrentDataParamsEdits ❌ `{"error":"'CommThread' object has no attribute 'getCurrentParamsEdits'"}`
> ```
> 
> The endpoint for editable values appears to be:
> 
> `http://econet_local_ip/econet/rmCurrentDataParamsEdits`
> 
> Let me know if you need help explaining what each endpoint does or how to use them.

Do you need the results of both Params-endpoints too?
I don't know how to use the _rm_ endpoints properly. Maybe which attributes I have to add.

Temporarily, I am using this script to change some simple parameters on the econet360i device in HA:
`rest_command:
  set_new_econet_param:
    url: "http://_econet_local_ip_/econet/newParam?newParamName={{ key }}&newParamValue={{ value }}&_=UID"
    method: GET
    username: _un_
    password: _pw_`

**Comment by arni077r** [comment]
When this `http://econet_local_ip/econet/rmCurrentDataParamsEdits`
is throwing error, try this
`http://econet_local_ip/econet/editParams`

**Comment by Rumb0ldt** [comment]
> econet/editParams ✔️ 

Nice! This command replies also a JSON with a lot of values.

**Comment by jontofront** [comment]
> > ```
> > http://econet_local_ip/econet/regParams ✔️ 
> > http://econet_local_ip/econet/regParamsData ❌ `{"error":"'CommThread' object has no attribute 'getCurrentParamsValues'"}`
> > http://econet_local_ip/econet/sysParams ✔️ 
> > http://econet_local_ip/econet/rmCurrentDataParams ❌  `{"error":"'CommThread' object has no attribute 'getRemoteMenuParams'"}`
> > http://econet_local_ip/econet/rmCurrentDataParamsEdits ❌ `{"error":"'CommThread' object has no attribute 'getCurrentParamsEdits'"}`
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > The endpoint for editable values appears to be:
> > `http://econet_local_ip/econet/rmCurrentDataParamsEdits`
> > Let me know if you need help explaining what each endpoint does or how to use them.
> 
> Do you need the results of both Params-endpoints too? I don't know how to use the _rm_ endpoints properly. Maybe which attributes I have to add.
> 
> Temporarily, I am using this script to change some simple parameters on the econet360i device in HA: `rest_command: set_new_econet_param: url: "http://_econet_local_ip_/econet/newParam?newParamName={{ key }}&newParamValue={{ value }}&_=UID" method: GET username: _un_ password: _pw_`

IF you give me script write post parameters I try implement it to code

**Comment by jontofront** [comment]
> When this `http://econet_local_ip/econet/rmCurrentDataParamsEdits` is throwing error, try this `http://econet_local_ip/econet/editParams`

so this link http://econet_local_ip/econet/editParams work? If works we can try make one of the feature

**Comment by Rumb0ldt** [comment]
> > When this `http://econet_local_ip/econet/rmCurrentDataParamsEdits` is throwing error, try this `http://econet_local_ip/econet/editParams`
> 
> so this link http://econet_local_ip/econet/editParams work? If works we can try make one of the feature

Yes. JSON is attached.

[editParams.json](https://github.com/user-attachments/files/21580761/editParams.json)

**Comment by Rumb0ldt** [comment]
> > Do you need the results of both Params-endpoints too? I don't know how to use the _rm_ endpoints properly. Maybe which attributes I have to add.
> > Temporarily, I am using this script to change some simple parameters on the econet360i device in HA: `rest_command: set_new_econet_param: url: "http://_econet_local_ip_/econet/newParam?newParamName={{ key }}&newParamValue={{ value }}&_=UID" method: GET username: _un_ password: _pw_`
> 
> IF you give me script write post parameters I try implement it to code

**Set Buffer Target Temp**
key: 183
value: [value]

**Set DHW Temp**
key: 103
value: [value]

**Edit Work function (Mode working)**
key: 162
value: 6 (Auto)
value: 2 (Winter)
value: 1 (Summer)

**Heat pump work mode (Mode working)**
key: 1231 (name: HP_work_state_set)
value: 1 (On)
value: 2 (Schedule)
value: 0 (Off)

Did you mean such combinations?

[regParams.json](https://github.com/user-attachments/files/21580793/regParams.json)
[sysParams.json](https://github.com/user-attachments/files/21580792/sysParams.json)

**Comment by cashford8** [comment]
Hello all

Firstly, great work on this integration @jontofront. I'm following it with interest!

I realise I'm tagging onto an existing feature request, but I believe it is the same device... however happy to open another ticket if you want.

I also have an ecomax360i connected to an air source heat pump. Home assistant grabs some of the temperatures but none of the other parameters:

<img width="499" height="824" alt="Image" src="https://github.com/user-attachments/assets/d36a3255-dcd0-4a7e-b783-ecdffdeea9fa" />

I would particularly be interested in the power sensors (electrical input and heat output - which only shows when the pump is running), COP efficiency, and also being able to run a hot water boost - all of these can be done via econet24.

<img width="1718" height="595" alt="Image" src="https://github.com/user-attachments/assets/3e6d8ffd-67b7-4ecf-bef7-7303a8db51e3" />

I can provide the json for regParams etc. - let me know what you need.

**Comment by jontofront** [comment]
> Hello all
> 
> Firstly, great work on this integration [@jontofront](https://github.com/jontofront). I'm following it with interest!
> 
> I realise I'm tagging onto an existing feature request, but I believe it is the same device... however happy to open another ticket if you want.
> 
> I also have an ecomax360i connected to an air source heat pump. Home assistant grabs some of the temperatures but none of the other parameters:
> 
> <img alt="Image" width="499" height="824" src="https://private-user-images.githubusercontent.com/96023652/476488651-d36a3255-dcd0-4a7e-b783-ecdffdeea9fa.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ5MDkxNzEsIm5iZiI6MTc1NDkwODg3MSwicGF0aCI6Ii85NjAyMzY1Mi80NzY0ODg2NTEtZDM2YTMyNTUtZGNkMC00YTdlLWI3ODMtZWNkZmZkZWVhOWZhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODExVDEwNDExMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAwNzcyODg5ZmEzODYyOGY3YjNmMGQ2MDc3Y2IzZTA3ODBmMjcxYWMyOTNhYTUzYTQ5NDRkMmE3ZGM1Y2Q4NDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.c0ToiU0eroroKoYuGv4dn36Y6JrJVXMXtPpxJiwzJKs">
> I would particularly be interested in the power sensors (electrical input and heat output - which only shows when the pump is running), COP efficiency, and also being able to run a hot water boost - all of these can be done via econet24.
> 
> <img alt="Image" width="1718" height="595" src="https://private-user-images.githubusercontent.com/96023652/476496043-3e6d8ffd-67b7-4ecf-bef7-7303a8db51e3.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ5MDkxNzEsIm5iZiI6MTc1NDkwODg3MSwicGF0aCI6Ii85NjAyMzY1Mi80NzY0OTYwNDMtM2U2ZDhmZmQtNjdiNy00ZWNmLWJlZjctNzMwM2E4ZGI1MWUzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODExVDEwNDExMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIzZDgxNjRjY2I1ODVkNjk5MDgzY2Q0NGRlMGNjMjE1NDlhNTI2NmIzMTI4YzM4MGQ1NzA1ZDNhYzdjYTU1ODImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.gCXvCwa1ZGfSb6yyLawyB2Y0tUVXy9QZg6k5kNtmTRY">
> I can provide the json for regParams etc. - let me know what you need.

Hi,

check these exaples if the json https://github.com/jontofront/ecoNET-300-Home-Assistant-Integration/tree/master/tests/fixtures/ecoMAX360 like ours please register new task sensors key. If no sensors in example please give me your http://econet_local_ip/econet/regParams  and http://econet_local_ip/econet/sysParams

**Comment by cashford8** [comment]
Hi @jontofront and all

I've made a bit of progress with this, looking at the various api endpoints. 

`sysParams` and `regParams` don't have the keys I'm interested in... but `editParams` does

here are the jsons from my ecomax360i 
[regParams.json](https://github.com/user-attachments/files/21852191/regParams.json)
[sysParams.json](https://github.com/user-attachments/files/21852192/sysParams.json)
and
[editParams.json](https://github.com/user-attachments/files/21852193/editParams.json)

in my case the following are the relevant sensors in `/econet/editParams`:
- informationParams['222'] is heat pump SCOP
- informationParams['221'] is heat pump COP
- informationParams['211'] is heat pump electrical power input (kW)
- informationParams['212'] is heat pump heating power output (kW)
- informationParams['203'] is periodic consumption (Wh)
- informationParams['204'] is total consumption (Wh)
The following are also given by editParams endpoint but are editable via` /econet/newParam?newParamName=115&newParamValue=1`
- data['238'].value is target day temperature
- data['239'].value is target night temperature
- data['115'].value is hot water boost - setting it to 1 starts the boost.
- data['273'].value is heat pump curve setting
- data['275'].value is heat pump curve shift

Going through this process I've managed to write my own restAPI sensor templates in home assistant as suggested by @Rumb0ldt so I'm good to go with my automations. But this might be useful information for the integration you are writing. 

All the best.

**Comment by jontofront** [comment]
> i [@jontofront](https://github.com/jontofront) and all
> 
> I've made a bit of progress with this, looking at the various api endpoints.
> 
> `sysParams` and `regParams` don't have the keys I'm interested in... but `editParams` does
> 
> here are the jsons from my ecomax360i [regParams.json](https://github.com/user-attachments/files/21852191/regParams.json) [sysParams.json](https://github.com/user-attachments/files/21852192/sysParams.json) and [editParams.json](https://github.com/user-attachments/files/21852193/editParams.json)
> 
> in my case the following are the relevant sensors in `/econet/editParams`:
> 
> * informationParams['222'] is heat pump SCOP
> * informationParams['221'] is heat pump COP
> * informationParams['211'] is heat pump electrical power input (kW)
> * informationParams['212'] is heat pump heating power output (kW)
> * informationParams['203'] is periodic consumption (Wh)
> * informationParams['204'] is total consumption (Wh)
>   The following are also given by editParams endpoint but are editable via` /econet/newParam?newParamName=115&newParamValue=1`
> * data['238'].value is target day temperature
> * data['239'].value is target night temperature
> * data['115'].value is hot water boost - setting it to 1 starts the boost.
> * data['273'].value is heat pump curve setting
> * data['275'].value is heat pump curve shift
> 
> Going through this process I've managed to write my own restAPI sensor templates in home assistant as suggested by [@Rumb0ldt](https://github.com/Rumb0ldt) so I'm good to go with my automations. But this might be useful information for the integration you are writing.
> 
> All the best.

So should I try to implement one of the more important parameters for you when I have some time, or will you play around with scripts yourselves?

**Comment by Rumb0ldt** [comment]
> So should I try to implement one of the more important parameters for you when I have some time, or will you play around with scripts yourselves?

I would be very happy if you could incorporate important attributes/sensors into the integration! Currently, it's tedious to control everything with scripts. Especially the switch between cooling and heating water in the summer would be better controlled via automation rather than manually.

**Comment by jontofront** [comment]
Hello! @Rumb0ldt @dschense @cashford8 @arni077r  

 I've released a new alpha version with **ecoMAX360 support**. Please test it and let me know if you can change the circuit temperature settings.

## New Sensors/Entities Added for ecoMAX360 Devices

- **Circuit 1-7 Day Temperature** (ComfortTemp) - Number entities for setting day mode temperatures
- **Circuit 1-7 Night Temperature** (EcoTemp) - Number entities for setting night mode temperatures
- **14 total new number entities** for temperature control

## Testing Request

I apologize that I cannot test this myself as I don't have an ecoMAX360 device at home. It would be greatly appreciated if you could:

- ✅ Test the new circuit temperature controls
- ✅ Check if the values can be changed properly
- ✅ Report any issues or errors you encounter
- ✅ Add additional logging if needed for debugging
- ✅ Provide diagnostic information if problems occur

## Thank You

Thank you for your patience and help with testing! Your feedback is invaluable for improving this integration.

---
*Version: 1.2.0-a-ecomax360 (Alpha)*

https://github.com/jontofront/ecoNET-300-Home-Assistant-Integration/releases/tag/1.2.0-a-ecomax360

**Comment by cashford8** [comment]
Hi @jontofront 

Thanks very much for this. I think I have installed the 1.2.0 alpha version.

<img width="362" height="95" alt="Image" src="https://github.com/user-attachments/assets/8362338c-e53f-4aad-acef-45772ecb885a" />

However, the only new entities I can see is heater mode: winter/summer/auto and fuel consumption.

<img width="464" height="836" alt="Image" src="https://github.com/user-attachments/assets/95fdc886-5d88-4a10-8b24-8f46909a61d6" />

and when I try to select a mode a message appears.

<img width="372" height="39" alt="Image" src="https://github.com/user-attachments/assets/e6503f53-0e5e-4dd4-a728-454aa72f6b05" />

the logs show this error:
```
This error originated from a custom integration.

Logger: custom_components.econet300.select
Source: custom_components/econet300/select.py:151
integration: ecoNET300 (documentation, issues)
First occurred: 07:49:35 (2 occurrences)
Last logged: 07:50:20

Failed to change heater mode to winter - API returned failure
Failed to change heater mode to auto - API returned failure
```
and
```
Logger: homeassistant.components.websocket_api.http.connection
Source: components/websocket_api/commands.py:317
integration: Home Assistant WebSocket API (documentation, issues)
First occurred: 07:49:35 (2 occurrences)
Last logged: 07:50:20

[139754741240576] Error during service call to select.select_option: Failed to change heater mode to winter
[139754741240576] Error during service call to select.select_option: Failed to change heater mode to auto
```


There aren't any other new sensors/entities for circuit temperatures etc. 

I installed the integration by selecting it from the drop down of "need a different version" from HACS and choosing version as shown:
<img width="371" height="626" alt="Image" src="https://github.com/user-attachments/assets/6991b6e3-9696-47d2-95d1-832f197908ad" />

Is there anything else I can do to check it's installed correctly or help troubleshoot?

**Comment by glynhudson** [comment]
Hi, thanks for improving support for ecoMAX360i. I've just tried running 1.2.0-a6, but I've not noticed any difference compared to 1.1.15 which I was running previously. I'm using the ecoMAX360i with a Grant HP290 heat pump

<img width="429" height="356" alt="Image" src="https://github.com/user-attachments/assets/9ce9f02c-3f11-498c-ae38-6665ccf900e9" />

<img width="859" height="930" alt="Image" src="https://github.com/user-attachments/assets/47a555bc-193f-4c3c-b70b-b5b38b8e8c3a" />

<img width="1256" height="930" alt="Image" src="https://github.com/user-attachments/assets/bb453074-0661-4b87-b8d1-e5cd09d5ad5c" />

**Comment by jontofront** [comment]
Sorry for the slight delay — I’m working on updating the code for the ecomax810 model after finding some interdependencies. As soon as I finish, I’ll return to your task. Thanks a lot for your patience

**Comment by LeeNuss** [comment]
I've been working on integrating some sensors, numbers and selects, based on _1.2.0-a-ecomax360_, in case anyone is interested: https://github.com/LeeNuss/ecoNET-300-Home-Assistant-Integration/tree/feature/sensors-inputs-ecomax360i
Still plenty of other controls and sensors to add and probably needs a good refactoring but @jontofront this might help you when you pick up the work on the branch again.

**Comment by jontofront** [comment]
> I've been working on integrating some sensors, numbers and selects, based on _1.2.0-a-ecomax360_, in case anyone is interested: https://github.com/LeeNuss/ecoNET-300-Home-Assistant-Integration/tree/feature/sensors-inputs-ecomax360i Still plenty of other controls and sensors to add and probably needs a good refactoring but [@jontofront](https://github.com/jontofront) this might help you when you pick up the work on the branch again.

@LeeNuss There are quite a few changes in my branch. I’m currently waiting to verify that the new v1.2.0alpha1 ([v1.2.0a1](https://github.com/jontofront/ecoNET-300-Home-Assistant-Integration/releases/tag/v1.2.0a1)
) works correctly. Once confirmed, I’ll refactor the code for ecomax360.

I also forgot to push the latest changes related to ecomax360.

Could you also add the new diagnostic file there?

**Comment by LeeNuss** [comment]
Great, I'll give the new release a spin.
I've since then dug a bit more into the api and summarised all exposed parameters here.
I marked all parameters if they are exposed to a local api endpoint (many of them are only readable via the cloud api endpoints).
However afaik from my testing all of the editable parameters can be modified via the local api even if they cannot be read.

https://gist.github.com/LeeNuss/accef83caf6c2294c0eb8e43867c04e7

here's an anker link to read only values, hard to find as it's a huge list
https://gist.github.com/LeeNuss/accef83caf6c2294c0eb8e43867c04e7#read-only-parameters

**Comment by strator-x** [comment]
Hello everyone,

I have the same controller (ecoMAX360i) on an air to water heat pump and i managed to integrate all sensors and all controls (even service parameters as i'm also developing custom controls for compressor frequency and fan speed). 

It is somehow messing with the econet24 cloud as the device became sometimes unavailable on it, but i can control everything locally via HA (and via HA cloud when away). I suspect it is mainly related to the response time of the controller on the editParams endpoint which is 4-5s and some collision when the cloud try to reach at the same time.

As a relatively new user of HA and even Github plus surely not a good coder my work is more like trial-and-error tinkering and I would greatly appreciate being able to contribute to this project to share my work.

I started from the 1.1.15 version of this integration - thanks a lot for this great work - but i made a lot of modification (mainly ugly ones) so i can't no longer call it a fork...

Please feel free to ask me anything to help add the full support of this controller to this integration.

<img width="1041" height="320" alt="Image" src="https://github.com/user-attachments/assets/ae929742-b613-4654-bdf2-fe048625e551" />

<img width="665" height="1169" alt="Image" src="https://github.com/user-attachments/assets/2db3c04d-8f33-4aaa-b238-922baf7fb0f3" />

<img width="331" height="1242" alt="Image" src="https://github.com/user-attachments/assets/f22dd885-85fe-4914-a369-e78138b4ce46" />

<img width="326" height="1244" alt="Image" src="https://github.com/user-attachments/assets/ebe54180-103b-4971-8067-da61894c37be" />

<img width="338" height="1010" alt="Image" src="https://github.com/user-attachments/assets/39eea2b0-51bc-4b99-aed4-5cb53a1370db" />

**Comment by glynhudson** [comment]
Nice work @strator-x ! It would be fantastic to get this integrated into this repo. Could you open a pull request?

**Comment by jontofront** [comment]
Hello, Nice that others are also taking care of ecoMAX360. If you still need development/expansion with ecoMAX360 in this integration, then add/attach the ecoMAX360 diagnostic file.

**Comment by bartmi08** [comment]
@strator-x  I’m especially interested in how you managed to make Home Assistant control the temperature using a slider.
I’ve been struggling with this for a long time and can’t figure out how to get Home Assistant to control the temperature for individual zones while still having automations work.
I also have an EcoMax 360, like you.
At the moment, I’m changing the temperature manually via econet24.com.

**Comment by gavyb** [comment]
Hi, just thought I would jump of this and offer some support in terms of testing. I may even be tempted to try some coding - I thought my coding days were well behind me!
I have a Grant Aerona 290 with an ecoMax360i controller and would really like to get more data exposed to allow me to capture stuff like energy consumption and to be able to control the temperature.
Here are the three files showing RegParams, sysParams and editParams:

[regParams.json](https://github.com/user-attachments/files/25445691/regParams.json) 
[sysParams.json](https://github.com/user-attachments/files/25445690/sysParams.json)
[editParams.json](https://github.com/user-attachments/files/25445692/editParams.json)

Currently (on ver 1.2.0) the following entities are created:

<img width="652" height="1714" alt="Image" src="https://github.com/user-attachments/assets/b719dc4c-839f-4f6e-8c20-d366ff158b17" />

I also have some Rest API's set up to expose the data as follows:

<img width="642" height="1434" alt="Image" src="https://github.com/user-attachments/assets/e658f97a-b2af-4bdb-b7b7-e1f67a319834" />

<img width="657" height="1564" alt="Image" src="https://github.com/user-attachments/assets/bee262af-fe2e-47b9-b094-5df805fa9f0c" />

<img width="643" height="1561" alt="Image" src="https://github.com/user-attachments/assets/b3786954-4888-469a-a8bb-6741204d8bc5" />

**Comment by LeeNuss** [comment]
@gavyb have a look if my project fits your requirements.
I've written a completely separate gateway service to the econet300 with dedicated Home Assistant integration on top. It exposes all parameters available from the controller, istead of the small subset that econet300 exposes to the local api.

[econext-gateway](https://github.com/LeeNuss/econext-gateway) -- a service that runs on a device with a USB-to-RS485 adapter connected to the controller. It handles the GM3 protocol (discovery, bus token handling, polling) and exposes a REST API.
[ecoNEXT](https://github.com/LeeNuss/econext) -- a HACS integration for Home Assistant that connects to the gateway api and provides large number of sensors and controls. Currently supports parameter discovery (~1800 params), reading, writing, alarm history, and has climate entities for each active circuit.
Setup instructions are in the repository or here on this [forum thread](https://community.openenergymonitor.org/t/econext-direct-local-control-of-grant-hp290-econet-replacement-plum-ecomax360i/29758)

**Comment by GerMarsh** [comment]
A big Welsh thank you to all who have contributed! We have just had installed a Grant Aerona 290 ASHP and I was so pleased to see the Econet 300 integration. I know more about MQTT than Python so I was hoping to publish the power consumption metrics via MQTT to HA. Then I found out that they are under Service Parameters and require a service password. It seems the world and his wife knows these so no problems there but attempting to work out the API endpoint is tricky. I first resorted to Wireshark but could see nothing coming back from the econet24 system. I then remembered my code for Beautiful Soup which I use for web page scraping on a Raspberry Pi. With that and Mechanize I managed to login to econet24 but then thought that this is getting complicated! We have a Shelly device monitoring the mains supply to the garage and basement where the power for the ASHP comes from. Using the HA integration's "Boiler Electrical Power" and the Shelly value I will just subtract 550W from the total and allocate that to the heat pump using a template. Like some others have tried, I just get errors with the endpoints which start with "rm". I assume that those are for the remote values whereby econet24 needs to send the passwords to get the current metric values. 

Sorry this has been a bit long! By the way, our "Smart controller" (ecoMAX360i) has no sensor connection for Circuit one. Econet24 just uses the "Heat Pump Actual Temperature" for the temp of Circuit 1. Circuit 2 has disappeared completely off that Current data screen too. If anyone has any information on the actual meaning of the Econet24 metrics I would love to see it. 

(I think that LeeNuss has done a cracking job with his Econet replacement! As I have a spare Rpi around here somewhere, I will give it a go.)

Thank you all again.

**Comment by chrismfield** [comment]
@strator-x your work looks great. Would love to get this working on my system with control for my system from HA. However I can't use your repo - can you send a pull request? @jontofront happy to share any diagnostic files - not sure where to get these from?

**Comment by strator-x** [comment]
Sorry for the very late follow-up here.

I finally cleaned up the work I had locally and opened a draft PR with the current working ecoMAX360i implementation:

#234

Thanks @glynnhudson for explicitly pushing me to open a PR, and thanks @Rumb0ldt / @dschense for the ecoMAX360 / endpoint details and examples in this issue. Also thanks @bartmi08 and @gayvb for the interest and testing/support offers around ecoMAX360i.

A big thanks to @jontofront for the base integration and the ongoing development/maintenance. This would not have been practical without that existing work.

The PR is intentionally opened as draft. The patch is quite large because it includes the full working implementation I tested locally on an ecoMAX360i controller, but I expect it may need to be split before merge.

Main things included in the draft:

* better local polling robustness and stale-data handling;
* health diagnostics such as live polling status, data age, consecutive failures, and last successful update timestamp;
* safer handling of partial/empty `sysParams` and `regParams`;
* `editParams` support for readable/writable parameters;
* safer number/select/switch entities for editable parameters;
* min/max fallback handling when controller metadata is incomplete;
* broader ecoMAX360i sensor exposure;
* Phoenix internal temperature scaling fix;
* avoiding unsupported generic heater/boiler controls when the ecoMAX360i rejects them.

My proposal is to use #234 first as a review/discussion branch, then split it if preferred, for example into:

1. coordinator health/stale diagnostics;
2. `editParams` read/write safety;
3. ecoMAX360i sensor coverage;
4. unsupported control hiding.

I am happy to adjust the PR structure depending on what maintainers prefer.

**Comment by gavyb** [comment]
Hi, I thought I would check on the status of the ecoMax360i support and provide some feedback.
I am getting a few errors related to entities without a state class - please see examples attached 

<img width="1080" height="2091" alt="Image" src="https://github.com/user-attachments/assets/0899e7be-e93d-4c8e-aae6-ee17fca57693" />

<img width="1080" height="2091" alt="Image" src="https://github.com/user-attachments/assets/323fa472-c446-4b90-a0dd-7cb5d399d2cf" />

<img width="1080" height="2091" alt="Image" src="https://github.com/user-attachments/assets/e24dd7ed-9d80-4715-ab3e-0838f1a8f6ad" />

Thanks.

**Comment by gavyb** [comment]
I would also like to get total energy for CH & DHW. I can't find these in editparams, regparams or sysparams, but can see them in ecoNET as follows:

<img width="2836" height="1534" alt="Image" src="https://github.com/user-attachments/assets/34b37aa0-472e-4640-9668-597477a19767" />

**Comment by GerMarsh** [comment]
I'm sorry if I am not following the github protocol but I too wanted the energy consumption as determined by the heat pump (Grant R290) and found the ecoNET integration was not providing all of them. However, I am quite astounded by the effort gone into that integration as I had no response from Plum when I asked for the specs!

I think, for example, that SCOP worked originally but perhaps there was some update on the heat pump and then it returns "unknown". 

After a bit of farting about I worked out that SCOP (as reported in ecoNET24) is under "informationParams" under the editParams endpoint. I have used the REST integration to implement that and a few others:

```
# Corrected on 2nd Sep 2026      
    - name: "ASHP CH Heating Energy"
      unique_id: 'information192'
      value_template: "{{ value_json['informationParams']['192'][1][0][0]}}"
      unit_of_measurement: "kWh"
      device_class: energy
      state_class: measurement

    - name: "ASHP DHW Heating Energy"
      unique_id: 'information191'
      value_template: "{{ value_json['informationParams']['191'][1][0][0]}}"
      unit_of_measurement: "kWh"
      device_class: energy
      state_class: measurement
      
#
# New at Sept 2026
#
    - name: "ASHP SCOP"
      unique_id: 'ASHP_SCOP'
      value_template: "{{ value_json['informationParams']['205'][1][0][0]}}"
      state_class: measurement
      
    - name: "ASHP Electrical Power"
      unique_id: 'information201'
      value_template: "{{ value_json['informationParams']['201'][1][0][0]}}"
      unit_of_measurement: "kW"
      device_class: power
      state_class: measurement
      
    - name: "ASHP Thermal Power"
      unique_id: 'information202'
      value_template: "{{ value_json['informationParams']['202'][1][0][0]}}"
      unit_of_measurement: "kW"
      device_class: power
      state_class: measurement
```
I think that the ecoNET integration's Electrical Power is actually Thermal Power whereas Periodic Energy Consumption - which suggests kWh - is the real Electrical Power.

I hope that the above makes sense as I am one of the old school and not used to these tools!

Nevertheless, I am so grateful for the time and expertise you have put into this integration! I was originally going to attempt to interrogate the controller and use MQTT to transfer the metrics. Thank you so much!!

**Comment by glynhudson** [comment]
Agree, the development of this integration is fantastic. 

Unfortunately the Grant onboard heat output figures are very incorrect, therefore the SCOP reading should be ignored. See comparison with indepnednat MID heat meters: https://community.openenergymonitor.org/t/grant-r290-inbuilt-monitoring-vs-mid-meters/30119. The onboard electric monitoring is reasonably correct.

**Comment by GerMarsh** [comment]
I found another incorrect endpoint call, this time for sensor.boiler_flow_rate. 

The working equivalent REST yaml is...
```
    - name: "ASHP Flow Rate"
      unique_id: 'information211'
      value_template: "{{ value_json['informationParams']['211'][1][0][0]}}"
      unit_of_measurement: "L/min"
      device_class: volume_flow_rate
      state_class: measurement
```
I see the point about the COP. I wonder how the heat pump calculates its heat output. I agree that the electrical consumption is quite accurate. I don't monitor the heat pump separately but do monitor the basement where it is located and the additional load seen is equal to that reported by the heat pump.

Thanks to the combination of this integration and the YAML code, my display now looks like this...

<img width="1700" height="952" alt="Image" src="https://github.com/user-attachments/assets/5ba5257c-cf0c-429d-909a-d6afa89400b5" />

**Comment by GerMarsh** [comment]
I assume that, as I cannot find any formal API or spec docs for the Plum device, there is always a risk that they could agree with Grant (or perhaps Axen) to insert a new metric using an existing parameter and move the original metrics down one! We could always be playing catch-up if that's the case.

Thank you for all your efforts. Even the installer is impressed!

**Comment by GerMarsh** [comment]
A few more bits of information I have found out. I apologise if this is not the correct protocol for these type of things!

1. The integration's "Periodic energy consumption" is, at the moment, much more accurate than ecoNET24's display which I think has hit a 16 bit limit with 32768.0!
2. I think the current flow rate for the fan needs to be multiplied by 100. According to the manual, the ASHP rated air flow for our heat pump (HPR29155) is 4050 cubic metres per hour.

It seems a few params have changed over the last few years, at least for our heat pump. Here are the REST entries for electrical consumption for CH and DHW:

```
# Corrected on 2nd Sep 2026      
    - name: "ASHP CH Heating Energy"
      unique_id: 'information192'
      value_template: "{{ value_json['informationParams']['192'][1][0][0]}}"
      unit_of_measurement: "kWh"
      device_class: energy
      state_class: measurement

    - name: "ASHP DHW Heating Energy"
      unique_id: 'information191'
      value_template: "{{ value_json['informationParams']['191'][1][0][0]}}"
      unit_of_measurement: "kWh"
      device_class: energy
      state_class: measurement
```
And here's the Template to calculate the total...

```
      - name: "ASHP Total Electrical Energy"
        unit_of_measurement: "kWh"
        state: >
           {{ (states("sensor.ashp_ch_heating_energy")|float + states("sensor.ashp_dhw_heating_energy")|float) |round(2) }}
```
Thank you all for your work on this. I hope to get the heat pump efficient before the Welsh winter hits!
